src/containers/GradebookPage/index.jsx

```diff
@@ -1,45 +1,83 @@
+/* eslint-disable import/no-named-as-default */
+import React from 'react';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+
+import { Tab, Tabs } from '@edx/paragon';
 
-import thunkActions from 'data/thunkActions';
-import actions from 'data/actions';
 import selectors from 'data/selectors';
+import thunkActions from 'data/thunkActions';
+
+import WithSidebar from 'components/WithSidebar';
+import GradebookHeader from 'components/GradebookHeader';
+import GradesTab from 'components/GradesTab';
+import GradebookFilters from 'components/GradebookFilters';
+import GradebookFiltersHeader from 'components/GradebookFiltersHeader';
+import BulkManagement from 'components/BulkManagement';
+
+export class GradebookPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.updateQueryParams = this.updateQueryParams.bind(this);
+  }
 
-import Gradebook from 'components/Gradebook';
-
-const mapStateToProps = (state, ownProps) => {
-  const {
-    root,
-    filters,
-    grades,
-  } = selectors;
-
-  const { courseId } = ownProps.match.params;
-  return {
-    courseId,
-    filteredUsersCount: grades.filteredUsersCount(state),
-    gradeExportUrl: root.gradeExportUrl(state, { courseId }),
-    interventionExportUrl: root.interventionExportUrl(state, { courseId }),
-    selectedTrack: filters.track(state),
-    selectedCohort: filters.cohort(state),
-    selectedAssignmentType: filters.assignmentType(state),
-    showBulkManagement: root.showBulkManagement(state, { courseId }),
-    showSpinner: root.shouldShowSpinner(state),
-    totalUsersCount: grades.totalUsersCount(state),
-  };
+  componentDidMount() {
+    const urlQuery = queryString.parse(this.props.location.search);
+    this.props.initializeApp(this.props.courseId, urlQuery);
+  }
+
+  updateQueryParams(queryParams) {
+    const parsed = queryString.parse(this.props.location.search);
+    Object.keys(queryParams).forEach((key) => {
+      if (queryParams[key]) {
+        parsed[key] = queryParams[key];
+      } else {
+        delete parsed[key];
+      }
+    });
+  }
+
+  render() {
+    return (
+      <WithSidebar
```

Comment:
Nit: I know we probably do this for layout reasons but I think it's weird to have a sidebar that's only logically connnected to the `GradesTab` at the `Page` level. Also the `WithSidebar` naming is a bit bizarre. I get what it's doing but my reaction is 🤨 every time I look at it.

Reply:
With<Thing> is a pretty common React paradigm.  
In comparison to `<Drawer content=<Main content>>{sidebar content}</Drawer>`, it felt much cleaner.

as for the sidebar only being related to one tab... that alas is an existing issue.